lgc/patch/FragColorExport.cpp

```diff
@@ -658,26 +658,41 @@ CallInst *LowerFragColorExport::AddExportForGenericOutputs(Function *fragEntryPo
       exportValues[hwColorTarget] = getOutputValue(expFragColors[hwColorTarget].value, location, insertPos);
   }
 
+  // Add the color export information to the palmetadata.
+  SmallVector<ColorExportInfo, 8> info;
+  for (unsigned hwColorTarget = 0; hwColorTarget < MaxColorTargets; ++hwColorTarget) {
+    Value *output = exportValues[hwColorTarget];
+    if (output == nullptr)
+      continue;
+    const ColorValueInfo &colorExportInfo = expFragColors[hwColorTarget];
+    info.push_back({hwColorTarget, colorExportInfo.location, colorExportInfo.isSigned, output->getType()});
+  }
+
+  if (info.empty()) {
+    m_pipelineState->getPalMetadata()->updateSpiShaderColFormat(info);
```

Comment:
I don't think I quite understand how you're dealing with an FS with no exports in the unlinked case. Are you doing this here so that case doesn't need an export glue shader? How does the dummy export instruction get added?

Reply:
A [later PR](https://github.com/GPUOpen-Drivers/llpc/pull/796/files#diff-d0beb3efdc3cced1f1d7f59508f2fa7eR622) this might become clearer.  If there are no exports to generate (unlinked case or not), then this function will return nullptr.  This tells [the caller](https://github.com/GPUOpen-Drivers/llpc/pull/774/files#diff-d0beb3efdc3cced1f1d7f59508f2fa7eR522) that no exports were added and it should add the dummy export if required.

For the unlinked case, the "exports" will be the return statement, so a non null value will be returned.  This is made explicit in a later PR.